xcube_stac/store.py

```diff
@@ -0,0 +1,337 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Any, Tuple, Iterator, Dict, Container, Union
+
+import logging
+import xarray as xr
+
+from xcube.util.jsonschema import (
+    JsonObjectSchema,
+    JsonStringSchema
+)
+from xcube.core.store import (
+    DATASET_TYPE,
+    DataDescriptor,
+    DataStore,
+    DataStoreError,
+    DataTypeLike
+)
+from .constants import DATASET_OPENER_ID
+from .opener import StacDataOpener
+from .stac import Stac
+
+_LOG = logging.getLogger("xcube")
+
+
+class StacDataStore(StacDataOpener, DataStore):
+    """ STAC implementation of the data store.
+
+    Attributes:
+        **stac_kwargs: Parameters required by the STAC data store.
+            * url: URL to STAC catalog (required)
+            * collection_prefix: Path of collection used as
+                entry point (optional)
+            * data_id_delimiter: Delimiter used to separate
+                collections, items and assets from each other (optional)
+    """
+
+    def __init__(self, **stac_kwargs):
+        super().__init__(stac=Stac(**stac_kwargs))
+
+    @classmethod
+    def get_data_store_params_schema(cls) -> JsonObjectSchema:
+        """ Get the JSON schema for instantiating a new data store.
```

Comment:
Please only repeat API docs from public parents if

* The implementation changes the described behaviour of the base class
* The specific implementation requires special documentation, e.g., because it is complex
* You want to list & document explicit keyword arguments which are `**kwargs` in the overridden method
* The base  class is not public or documented and can not be pointed to

Reply:
Note, that we always try to avoid redundancy. Like code duplication, documentation duplication makes code harder to maintain, e.g., consider API doc updates in the base class.

Reply:
For easy functions, I removed the API docs. For more complex functions, I kept them and will reconsider when I implement them. Thanks for the heads up!